ninjasworkout/app.js

```diff
@@ -0,0 +1,164 @@
+var session = require('express-session');
+
+
+const express = require('express');
+const app = express();
+const path = require('path');
+var bodyParser = require('body-parser');
+var mongoose = require('mongoose');
+const dbConfig = require('./database/db');
+var cookieparser =require('cookie-parser');
+const csurf = require('csurf');
+var multer = require('multer');
+var upload = multer();
+const rateLimit = require("express-rate-limit");
+const fileupload = require("express-fileupload");
+//app.use('/images', express.static(__dirname + '/images'));
+blinker = require("express-blinker"); //directory traversal
+basePath = path.join(__dirname, "images");
+console.log(basePath)
+const limiter = rateLimit({
+    max: 30,
+    windowMs: 1 * 60 *1000 ,
+    //message:"sorry "
+    
+})
+
+app.use(blinker(basePath, [
+  {
+      test: /.*/,
+      etag: true,
+      lastModified: false,
+      cacheControl: true,
+      expires: false,
+      age: 600
+  }
+]));
+
+
+
+const imageStorage = multer.diskStorage({
+    // Destination to store image     
+    destination: 'images', 
+      filename: (req, file, cb) => {
+       // console.log(file.originalname)
+        //console.log("fil extension"+ path.extname(file.originalname))
+
+          cb(null, file.fieldname + '_' + Date.now() 
+             + path.extname(file.originalname))
+            // file.fieldname is name of the field (image)
+            // path.extname get the uploaded file extension
+    }
+  });
+  
+  
+  const imageUpload = multer({
+    storage: imageStorage,
+    limits: {
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 0d79a4cae3d090d5360737c4b7aa7b0d -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Uncontrolled Resource Consumption
_High CWE-400_
The `multer` module does not restrict the number of file/non-file fields in a single request. Thus, a denial-of-service attack is possible, because an attacker may submit a very large number of file/non-file fields in one request.



### How to fix
Potential mitigations include the following:
  1. Design throttling mechanisms into the system architecture. The best protection is to limit the amount of resources that an unauthorized user can cause to be expended. A strong authentication and access control model will help prevent such attacks from occurring in the first place. The login application should be protected against DoS attacks as much as possible. Limiting the database access, perhaps by caching result sets, can help minimize the resources expended. To further limit the potential for a DoS attack, consider tracking the rate of requests received from users and blocking requests that exceed a defined rate threshold.
  2. Mitigation of resource exhaustion attacks requires that the target system either:

    * recognizes the attack and denies that user further access for a given amount of time, or
    * uniformly throttles all requests in order to make it more difficult to consume resources more quickly than they can again be freed.

The first of these solutions is an issue in itself though, since it may allow attackers to prevent the use of the system by a particular valid user. If the attacker impersonates the valid user, they may be able to prevent the user from accessing the server in question.

The second solution is simply difficult to effectively institute -- and even when properly done, it does not provide a full solution. It simply makes the attack require more resources on the part of the attacker.

  3. Ensure that protocols have specific limits of scale placed on them.
  4. Ensure that all failures in resource allocation place the system into a safe posture.